src/main/java/mekanism/common/tile/prefab/TileEntityDoubleElectricMachine.java

```diff
@@ -0,0 +1,287 @@
+package mekanism.common.tile.prefab;
+
+import mekanism.api.EnumColor;
+import mekanism.api.transmitters.TransmissionType;
+import mekanism.common.MekanismBlocks;
+import mekanism.common.MekanismItems;
+import mekanism.common.SideData;
+import mekanism.common.Tier.BaseTier;
+import mekanism.common.Upgrade;
+import mekanism.common.base.IFactory.RecipeType;
+import mekanism.common.base.ITierUpgradeable;
+import mekanism.common.recipe.RecipeHandler;
+import mekanism.common.recipe.inputs.DoubleMachineInput;
+import mekanism.common.recipe.machines.DoubleMachineRecipe;
+import mekanism.common.recipe.outputs.ItemStackOutput;
+import mekanism.common.tile.TileEntityFactory;
+import mekanism.common.tile.component.TileComponentConfig;
+import mekanism.common.tile.component.TileComponentEjector;
+import mekanism.common.util.ChargeUtils;
+import mekanism.common.util.InventoryUtils;
+import mekanism.common.util.MekanismUtils;
+import mekanism.common.util.MekanismUtils.ResourceType;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.NonNullList;
+
+public abstract class TileEntityDoubleElectricMachine<RECIPE extends DoubleMachineRecipe<RECIPE>> extends TileEntityBasicMachine<DoubleMachineInput, ItemStackOutput, RECIPE> implements ITierUpgradeable
+{
+	/**
+	 * Double Electric Machine -- a machine like this has a total of 4 slots. Input slot (0), secondary slot (1), output slot (2),
+	 * energy slot (3), and the upgrade slot (4). The machine will not run if it does not have enough electricity.
+	 *
+	 * @param soundPath - location of the sound effect
+	 * @param name - full name of this machine
+	 * @param maxEnergy - maximum amount of energy this machine can hold.
+	 * @param baseEnergyUsage - how much energy this machine uses per tick.
+	 * @param ticksRequired - how many ticks it takes to smelt an item.
+	 */
+	public TileEntityDoubleElectricMachine(String soundPath, String name, double maxEnergy, double baseEnergyUsage, int ticksRequired)
+	{
+		super(soundPath, name, maxEnergy, baseEnergyUsage, 4, ticksRequired, MekanismUtils.getResource(ResourceType.GUI, "guibasicmachine.png"));
```

Comment:
On the combiner today, this is [guiadvancedmachine.png](https://github.com/aidancbrady/Mekanism/blob/89e10b1d9d0399ca821c30db3f416638522ee723/src/main/resources/assets/mekanism/gui/guiadvancedmachine.png), which has a bar between the cobblestone and dust slots that showed the liquid stone level. I'm not sure whether it'd be better to change that to [guibasicmachine.png](https://github.com/aidancbrady/Mekanism/blob/89e10b1d9d0399ca821c30db3f416638522ee723/src/main/resources/assets/mekanism/gui/guibasicmachine.png) as I did here, which replaces the bar with an arrow pointing up from the cobblestone to the dust, or to [guiblank.png](https://github.com/aidancbrady/Mekanism/blob/89e10b1d9d0399ca821c30db3f416638522ee723/src/main/resources/assets/mekanism/gui/guiblank.png), which would just remove the bar.

Reply:
yeah `basicmachine` should be fine